src/citrine/seeding/find_or_create.py

```diff
@@ -147,6 +147,10 @@ def create_or_update(*, collection: Collection[CreationType], resource: Creation
         # Copy so that passed-in resource is unaffected
         new_resource = deepcopy(resource)
         new_resource.uid = old_resource.uid
+        # This is only relevant for locally created design workflows, since
+        # they likely won't have a branch ID but need one to be updated.
+        if hasattr(old_resource, "branch_id") and new_resource.branch_id is None:
```

Comment:
Can we do this with class methods instead of python introspection?  Mixins?

Reply:
I'm not super clear on what you're thinking. Care to elaborate?

As I see it, this is the only place we need to copy the `branch_id`, and checking the existence of the field instead of the type future-proofs it. And since it's the only place, propagating some method through the resources feels like overkill. But maybe I'm missing an obvious better way.

(PS I do need to tweak this line to check that `new_resource` has `branch_id` before confirming it's unset).

Reply:
Ah, just noticed your overall comment now (I always forget to check that). Yes, I can use `isinstance`, I was just defending against future resources with this pattern. But sure, I'll make that tweak.